### ISSUE TITLE
CPU usage

### ISSUE THREAD

**Issue by happysurf** [inbasic/turbo-download-manager#16]
First of all congratulations for this great tool.
I have noticed a 30% of CPU usage during download.
I think this is too much.
Firefox 42 + XP Pro + TDM 0.1.10.

**Comment by inbasic** [comment]
Can you try the latest dev version and let me know if the performance is acceptable or not?

https://github.com/inbasic/turbo-download-manager/blob/master/builds/packed/firefox.xpi?raw=true
Download the XPI and drop it into your Firefox

**Comment by happysurf** [comment]
Much better, now the average is about 10 % with 3 threads, for that the overall download speed is lower than official version. 

EDIT: An other thing, the icon on add.ons manager page is missing.
Thanks.

**Comment by inbasic** [comment]
> Much better, now the average is about 10 % with 3 threads, for that the overall download speed is lower than official version.

Can you compare the usage with dTa with the same number of threads?

> EDIT: An other thing, the icon on add.ons manager page is missing.

Thanks, will be fixed.

**Comment by happysurf** [comment]
With the same download source with 3 threads:
TDM 0.1.10 (official) = CPU use average 20 % (0 - 30%)
TDM 0.1.9 (send to me) = CPU use average 10 % (0 - 20%)

**Comment by inbasic** [comment]
the dev version looks good to me. Can you compare its performance with dTa as well https://addons.mozilla.org/fr/firefox/addon/downthemall/

**Comment by happysurf** [comment]
OK, same source with 3 threads:
TDM 0.1.9 (send to me) = CPU usage average 10 % (0 - 20%);
download sped: 400 - 500 kb/s.

DTA 2.0.18.1 = CPU usage average 2 % (0 - 5 %)
download sped: 400 - 500 kb/s.

The speed download is the same but DTA have a good CPU usage, practically is the same of Free Download Manager program.

In any case I have in my office an old PC with Intel Core Duo 8500.

**Comment by inbasic** [comment]
There are some more improvements. Can you try these custom builds and let me know their performance compared to the one linked above
1. https://www.dropbox.com/s/p4rcsgmzz4yjand/firefox-0k.xpi?dl=0
2. https://www.dropbox.com/s/l4zgaor6kmx0dno/firefox-100k.xpi?dl=0
3. https://www.dropbox.com/s/aunsb8jbh78k6eu/firefox-200k.xpi?dl=0
4. https://www.dropbox.com/s/u6k8ioaw2fcwylp/firefox-500k.xpi?dl=0

Basically in the above versions, writing to physical disk is limited to 0 bytes, 100KBytes, 200KBytes, and 500KBytes. My guess is the current CPU usage you have reported is mainly due to the high numbers of disk writing requests.

**Comment by happysurf** [comment]
OK, let's go. 
Same source with 6 thread:

ITD dev.  = min CPU: 5%, max CPU: 20%,  most frequent value: 15% - speed 850 KB/s

ITD 0      = min CPU: 0%, max CPU: 20%,  most frequent value: 10% - speed 850 KB/s
ITD 100  = min CPU: 0%, max CPU: 15%,  most frequent value:   5% - speed 850 KB/s
ITD 200  = min CPU: 0%, max CPU: 13%,  most frequent value:   3% - speed 850 KB/s
ITD 500  = min CPU: 0%, max CPU: 15%,  most frequent value:   4% - speed 850 KB/s

DTA        = min CPU: 2%, max CPU: 5%,    most frequent value:   4% - speed 650 KB/s

Great work, the speed is better than DTA and the CPU usage is good now.

**Comment by inbasic** [comment]
Perfect, let's stick with the 200K chunk size then.

**Comment by happysurf** [comment]
OK, I wait the official release.
Thanks.

**Comment by inbasic** [comment]
Version 0.1.11 has been pushed to AMO,
https://addons.mozilla.org/en-US/firefox/addon/turbo-download-manager/versions/
